theme/static/js/hs_file_browser.js

```diff
@@ -1277,8 +1341,22 @@ function refreshFileBrowser(name) {
                 }
             }
         }
-
         updateSelectionMenuContext();
+        $.ajax({
+            type: "GET",
+            url: '/hsapi/resource/' + SHORT_ID + '/file_list/',
+        }).complete(function(res) {
+            let pathObjs = JSON.parse(res.responseText).results
+
+            // conditionally render edit pencil control
+            if (RESOURCE_MODE === 'Edit') {
+                if (anyExternalFiles(pathObjs)) {
```

Comment:
Why do you need to do all of this?  Isn't all the information you need provided by the external_content key you added to the page processor?  This seems like it's incredibly costly and duplicates the work done by that key.

Reply:
After the asynchronous file deletion is completed, another server evaluation is performed to get the definitive state. The current Content manager code does a partial refresh of some of the page state and UI elements, but not others. The initial server context/JS passthrough at that point is stale.

Reply:
Once we move towards a more RESTful design, and stop with the tightly coupled Django/processor_for architecture, all of this will be easier. For now, it's a bit of a minefield.

Reply:
What's wrong with adding an endpoint that can check for referenced content for you that you can call?   Seems a lot easier to do than what I'm seeing here.  Also, this logic doesn't accurately check for referenced content... it just looks for files that end with .url

Reply:
Scott's comment here is in line with my comment above regarding checking things on the server side rather than doing it on the client side. Should be an easy check on the server side via an internal endpoint or passing in as a context variable.

Reply:
New endpoint